test/lib/fhirValidatorTest.ts

```diff
@@ -629,68 +629,88 @@ describe('fhir validator', () => {
         expect(err).to.not.be.null;
         expect(err instanceof Error).to.be.true;
         expect(Array.isArray(err.errors)).to.be.true;
         expect(err.toString()).to.contain('No resource provided');
-        done();
       }
     });
 
-    it('fails to validate with an explicit message if a non-string, non-object parameter is provided', done => {
+    it('fails to validate with an explicit message if a non-string, non-object parameter is provided', async () => {
       try {
         // @ts-ignore
-        fhirValidator.validate([]);
-        done();
+        await fhirValidator.validate([]);
       } catch (err) {
         expect(err).to.not.be.null;
         expect(err instanceof Error).to.be.true;
         expect(Array.isArray(err.errors)).to.be.true;
         expect(err.toString()).to.contain('Resource needs to be an object.');
-        done();
       }
     });
 
-    it('fails to validate with an explicit message if a parameter without a resourceType property is provided', done => {
+    it('fails to validate with an explicit message if a parameter without a resourceType property is provided', async () => {
       try {
-        fhirValidator.validate({
+        await fhirValidator.validate({
           // @ts-ignore
           uselessProperty: 'not a resource type',
         });
-        done();
       } catch (err) {
         expect(err).to.not.be.null;
         expect(err instanceof Error).to.be.true;
         expect(Array.isArray(err.errors)).to.be.true;
         expect(err.toString()).to.contain('Resource object does not have a resource type.');
         expect(err.toString()).to.not.contain('Did you mean to submit the .fhirResource property?');
-        done();
       }
     });
 
-    it('fails to validate with an explicit message if a parameter without a resourceType property is provided but a fhirResource property exists on the object', done => {
+    it('fails to validate with an explicit message if a parameter without a resourceType property is provided but a fhirResource property exists on the object', async () => {
       try {
-        fhirValidator.validate({
+        await fhirValidator.validate({
           // @ts-ignore
           uselessProperty: 'not a resource type',
           fhirResource: {
             why: 'You might want to submit me though',
           },
         });
-        done();
       } catch (err) {
         expect(err).to.not.be.null;
         expect(err instanceof Error).to.be.true;
         expect(Array.isArray(err.errors)).to.be.true;
         expect(err.toString()).to.contain('Resource object does not have a resource type.');
         expect(err.toString()).to.contain('Did you mean to submit the .fhirResource property?');
-        done();
+      }
+    });
+
+    it('fails to validate with an explicit message if a resource contains a resource that contains another resource', async () => {
+      try {
+        await fhirValidator.validate({
+          resourceType: 'DocumentReference',
```

Comment:
This resource is actually invalid for several reason, meaning that it might not check the right thing (the validator might object to other issues). In particular, there are required fields missing in the resources and general spec rules requires that any contained resource be referenced at least once. Would be nice to have a test resource that is invalid only for the reason being checked,

Reply:
That is true and that is why we check for the contain-contain case quite soon (before basics like "do you even have a resourceType") and return that as an error and in THIS test we compare the specific error message.

Reply:
Fair point. If you want one anyway, I wanted an example for my collection, too, and constructed the below which (per standard validator) only throws the double-contain-error:

```json
{
    "resourceType": "Observation",
    "contained": [{
        "resourceType": "Patient",
        "id": "pat",
        "contained": [{
            "resourceType": "Organization",
            "id": "org",
            "name": "Test org"

        }],
        "managingOrganization": {
            "reference": "#org"
        }
    }],
    "status": "final",
    "code": {
        "text": "example"
    },
    "subject": {
        "reference": "#pat"
    }
}
```